src/pybind/mgr/dashboard/controllers/cephfs.py

```diff
@@ -639,6 +639,17 @@ def rm_snapshot(self, fs_id, path, name):
         cfs = self._cephfs_instance(fs_id)
         cfs.rm_snapshot(path, name)
 
+    @RESTController.Resource('PUT', path='/_rename')
```

Comment:
what about `PUT /tree` since we already have endpoints for creating and removing a directory?
```suggestion
    @RESTController.Resource('PUT', path='/tree')
```

Reply:
rename already used by cephfs volume, I have not think of a better name. Do you mean we use ` PUT /tree` to represent the rename file and directory function?

Reply:
good idea

Reply:
but, we can also rename file, will this make users feel confused?

Reply:
@YiteGu yeah it could be confusing since tree usually refers to the directory.. 
and @dparmar18 ideally the api names should be simple, something the user can remember and easy to type so using underscores could be inconvenient for API names. so we can combine the file_or_dir under path and call this endpoint `rename-path`? I think path will cover both directory and file.

Reply:
https://github.com/ceph/ceph/blob/1272408fafd0d87fa8a3dd308b77e8f8867b8314/src/pybind/mgr/dashboard/controllers/cephfs.py#L94

Is using `rename_volume` more appropriate for this function?

Reply:
> that would create more changes for you (adapting the UI and test changes...) which I wouldn't recommend.

Got it.

Reply:
> @YiteGu yeah it could be confusing since tree usually refers to the directory.. and @dparmar18 ideally the api names should be simple, something the user can remember and easy to type so using underscores could be inconvenient for API names. so we can combine the file_or_dir under path and call this endpoint `rename-path`? I think path will cover both directory and file.

 agree with you. how do you think? @dparmar18

Reply:
```
Exception occurred:
  File "/home/docs/checkouts/readthedocs.org/user_builds/ceph/checkouts/58386/src/pybind/mgr/dashboard/plugins/feature_toggles.py", line 9, in <module>
    from ..controllers.cephfs import CephFS
  File "/home/docs/checkouts/readthedocs.org/user_builds/ceph/checkouts/58386/src/pybind/mgr/dashboard/controllers/cephfs.py", line 643
    def rename-path(self, fs_id, src_path, dst_path) -> None:
              ^
SyntaxError: invalid syntax
```
can not use `-` in function name of python. @nizamial09